### PR TITLE
Removed Delay from Button Press

### PR DESCRIPTION
**Pull request by Addehman**

### CODE REVIEW

File: Assets/Scripts/Puzzle/PressureButton.cs [hunk 1]
```diff
@@ -3,70 +3,41 @@
...
-		if (delayTimer > delayEnd) {
-			PlaySound();
-			colliderUp.enabled = false;
-			animator.SetBool("IsPressed", true);
-			IsActivated = true;
-			isDelayButtonDown = false;
-		}
+		PlaySound();
+		animator.SetBool("IsPressed", true);
+		IsActivated = true;
```
Comment (smeas) [inline Assets/Scripts/Puzzle/PressureButton.cs:31]:
```suggestion
		if (IsActivated) return;

		PlaySound();
		animator.SetBool("IsPressed", true);
		IsActivated = true;
```

File: Assets/Scripts/Puzzle/PressureButton.cs [hunk 2]
```diff
@@ -3,70 +3,41 @@
...
-		if (delayTimer > delayEnd) {
-			PlaySound();
-			colliderUp.enabled = true;
-			animator.SetBool("IsPressed", false);
-			IsActivated = false;
-			isDelayButtonUp = false;
-		}
+		PlaySound();
+		animator.SetBool("IsPressed", false);
+		IsActivated = false;
```
Comment (smeas) [inline Assets/Scripts/Puzzle/PressureButton.cs:38]:
```suggestion
		if (!IsActivated) return;

		PlaySound();
		animator.SetBool("IsPressed", false);
		IsActivated = false;
```

**Review by smeas — CHANGES_REQUESTED** [review]
Nice. Just a small bug. If you stand close to the button without pressing it and then walk away, it makes the click sound. You probably want to check `IsActivated` before playing the sound.

**Review by smeas — APPROVED** [review]

**Review by MrJackey — APPROVED** [review]
The animator bools could be optimized but feels good ingame 👍

**Pull request merged**